lib/validate-selectors.js

```diff
@@ -2,6 +2,56 @@ var listSequences = require('./list-sequences');
 var shouldIgnoreRule = require('./should-ignore-rule');
 var shouldIgnoreSelector = require('./should-ignore-selector');
 var toInterpoloatedRegexp = require('./to-interpolated-regexp');
+var resolveNestedSelector = require('postcss-resolve-nested-selector');
+
+function isTopLevel(node) {
+  return node.type === 'root';
+}
+
+function isNestedRule(node) {
+  return /(?:at)?rule/.test(node.parent.type)
+}
+
+function hasNoDeclarations(node) {
+  return node.nodes.length && node.every(function(child) {
```

Comment:
It fails if I use `||`. The main idea here is to skip rulesets that have no children at all:

``` css
.Component { 
  &--modifier {} 
}
```

Neither of these have declarations, but I'm only interested in the rule that contains another rule. The `&--modifier` ruleset still needs to be validated so I let that through.

Hope that makes sense.